app/features-json/api_controller.rb

```diff
@@ -98,6 +98,12 @@ def jwt
         authorization = request.env["HTTP_AUTHORIZATION"]
         bearer_token = authorization && authorization.slice(7..-1) # strip off the `Bearer `
 
+        # give the option to pass the bearer token as a query param.
+        # this is used when making websocket connections which do not allow us to set http headers
+        if bearer_token.nil?
```

Comment:
If we check for `nil` you can use `bearer_token ||= something`

Reply:
We could write it as this and get rid of the `if` condition here:

```ruby
bearer_token = authorization&.slice(7..-1) || request.params["bearer_token"]
```

Reply:
It would make the code harder to read though

Reply:
I think this is easier to read personally, since you're only setting the `bearer_token` in one place in the code.

```ruby
bearer_token = authorization && authorization.slice(7..-1)

if bearer_token.nil?
  bearer_token = request.params["bearer_token"]
end
```
> Current Implementation

___

```ruby
bearer_token = authorization&.slice(7..-1) || request.params["bearer_token"]
```
> Suggested Implementation

___

The way I read this is: slice the `authorization` if it exists, else `request.params["bearer_token"]`.

Is this what you're suggesting as more readable alternative?

```ruby
bearer_token = authorization && authorization.slice(7..-1)
bearer_token ||= request.params["bearer_token"]
```